tests/Doctrine/Tests/Persistence/Mapping/ColocatedMappingDriverTest.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Doctrine\Tests\Persistence\Mapping;
+
+use Doctrine\Persistence\Mapping\ClassMetadata;
+use Doctrine\Persistence\Mapping\Driver\ColocatedMappingDriver;
+use Doctrine\Persistence\Mapping\Driver\MappingDriver;
+use Doctrine\TestClass;
+use Generator;
+use PHPUnit\Framework\TestCase;
+
+class ColocatedMappingDriverTest extends TestCase
+{
+    /**
+     * @dataProvider pathProvider
+     */
+    public function testGetAllClassNames(string $path): void
+    {
+        $driver = new MyDriver($path);
```

Comment:
Shall we move the constructor call into a protected method that `AnnotationDriverTest` can override?